src/session/__tests__/sessionReducer-test.js

```diff
@@ -38,4 +60,149 @@ describe('sessionReducer', () => {
     const newState = sessionReducer(baseState, eg.action.login_success);
     expect(newState).toEqual({ ...baseState, needsInitialFetch: true });
   });
+
+  test('DEAD_QUEUE', () => {
+    const state = deepFreeze({
+      ...baseState,
+      needsInitialFetch: false,
+      loading: true,
+    });
+    const newState = sessionReducer(state, deepFreeze({ type: DEAD_QUEUE }));
+    expect(newState).toEqual({ ...baseState, needsInitialFetch: true });
+  });
+
+  test('LOGOUT', () => {
+    const state = deepFreeze({
+      ...baseState,
+      lastNarrow: [],
+      needsInitialFetch: true,
+      loading: true,
+    });
+    const newState = sessionReducer(state, deepFreeze({ type: LOGOUT }));
+    expect(newState).toEqual(baseState);
+  });
+
+  test('REALM_INIT', () => {
+    const action = deepFreeze({
+      ...eg.action.realm_init,
+      data: { ...eg.action.realm_init.data, queue_id: 100 },
+    });
+    const newState = sessionReducer(baseState, action);
+    expect(newState).toEqual({ ...baseState, eventQueueId: 100 });
+  });
+
+  test('DO_NARROW', () => {
+    const state = deepFreeze({
+      ...baseState,
+      lastNarrow: [],
+    });
+    const action = deepFreeze({
+      type: DO_NARROW,
+      narrow: privateNarrow('a@a.com'),
+    });
```

Comment:
formatting nit: Here and on a bunch of others where the state and/or action are quite simple expressions, it'd be helpful to squash them on one line. Just makes things closer together, and easier to scan through.

Quick keyboard tip for making that change in VS Code: I go to the end of the first line (after the `{`), and hit delete, end, delete, end, delete, then Ctrl+S to save-and-format.